plugins/aladino/functions/hasRequiredApprovals.go

```diff
@@ -0,0 +1,73 @@
+package plugins_aladino_functions
+
+import (
+	"fmt"
+
+	"github.com/reviewpad/reviewpad/v3/codehost/github/target"
+	"github.com/reviewpad/reviewpad/v3/handler"
+	"github.com/reviewpad/reviewpad/v3/lang/aladino"
+)
+
+func HasRequiredApprovals() *aladino.BuiltInFunction {
+	return &aladino.BuiltInFunction{
+		Type:           aladino.BuildFunctionType([]aladino.Type{aladino.BuildIntType(), aladino.BuildArrayOfType(aladino.BuildStringType())}, aladino.BuildBoolType()),
+		Code:           hasRequiredApprovalsCode,
+		SupportedKinds: []handler.TargetEntityKind{handler.PullRequest},
+	}
+}
+
+func hasRequiredApprovalsCode(e aladino.Env, args []aladino.Value) (aladino.Value, error) {
+	target := e.GetTarget().(*target.PullRequestTarget)
+	totalRequiredApprovals := args[0].(*aladino.IntValue).Val
+	requiredApprovalsFrom := args[1].(*aladino.ArrayValue).Vals
+
+	if len(requiredApprovalsFrom) < totalRequiredApprovals {
+		return nil, fmt.Errorf("hasRequiredApprovals: the number of required approvals exceeds the number of members from the given list of required approvals")
+	}
+
+	approvedBy, err := getApprovals(target)
+	if err != nil {
+		return nil, err
+	}
+
+	var requiredApprovalsCount int
+	for _, requiredApproval := range requiredApprovalsFrom {
+		if contains(approvedBy, requiredApproval.(*aladino.StringValue).Val) {
+			requiredApprovalsCount++
+		}
+
+		if requiredApprovalsCount == totalRequiredApprovals {
+			return aladino.BuildBoolValue(true), nil
+		}
+	}
+
+	return aladino.BuildBoolValue(false), nil
```

Comment:
I would suggest a different approach. IMO, the code becomes more readable, although the number of operations is higher. WDYT?

```suggestion
	var requiredApprovalsCount int
	for _, requiredApproval := range requiredApprovalsFrom {
		if contains(approvedBy, requiredApproval.(*aladino.StringValue).Val) {
			requiredApprovalsCount++
		}
	}

	return aladino.BuildBoolValue(requiredApprovalsCount > requiredApprovalsCount), nil
```